src/jackdaw/test/transports/kafka.clj

```diff
@@ -0,0 +1,201 @@
+(ns jackdaw.test.transports.kafka
+  (:require
+   [clojure.core.async :as async]
+   [clojure.tools.logging :as log]
+   [jackdaw.client :as kafka]
+   [jackdaw.data :as jd]
+   [jackdaw.test.commands :as cmd]
+   [jackdaw.test.transports :as t]
+   [jackdaw.test.serde :refer [apply-serializers apply-deserializers
+                               serde-map
+                               byte-array-serde]])
+  (:import
+   org.apache.kafka.streams.KafkaStreams$StateListener
+   org.apache.kafka.clients.consumer.ConsumerRecord
+   org.apache.kafka.clients.producer.ProducerRecord))
+
+(defn subscribe
+  "Subscribes to specified topics
+
+   `consumer` should be a kafka consumer
+   `topic-config` should be a sequence of topic-metadata maps"
+  [consumer topic-config]
+  (kafka/subscribe consumer topic-config))
+
+(defn load-assignments
+  [consumer]
+  (.poll consumer 0)
+  (.assignment consumer))
+
+(defn seek-to-end
+  "Seeks to the end of all the partitions assigned to the given consumer
+   and returns the updated consumer"
+  [consumer & topic-partitions]
+  (let [assigned-partitions (or topic-partitions (load-assignments consumer))]
+    (.seekToEnd consumer assigned-partitions)
+    (doseq [assigned-partition assigned-partitions]
+      ;; This forces the seek to happen now
+      (.position consumer assigned-partition))
+    consumer))
+
+(defn poller
+  "Returns a function that takes a consumer and puts any messages retrieved
+   by polling it onto the supplied `messages` channel"
+  [messages]
+  (fn [consumer]
+    (try
+      (let [m (.poll consumer 1000)]
+        (when m
+          (doseq [msg m]
+            (async/onto-chan messages [msg] false))))
+      (catch Throwable e
+        (log/error (Throwable->map e))
+        (async/put! messages {:error e})))))
+
+(defn subscription
+  "Subscribes to `topic-collection` and seeks to the end of all partitions. This
+   is usually what you want in a testing context. It's best for the test you're
+   trying to run now to ignore all the garbage created by previous tests."
+  [kafka-config topic-collection]
+  (-> (kafka/consumer kafka-config byte-array-serde)
+      (subscribe topic-collection)
+      (seek-to-end)))
```

Comment:
This is similar to `jackdaw.client/subscribed-consumer`:
https://github.com/FundingCircle/jackdaw/blob/master/src/jackdaw/client.clj#L141